src/aiida/engine/daemon/client.py

```
@@ -944,36 +970,39 @@ def _start_daemon(self, number_workers: int = 1, foreground: bool = False) -> No
         # Start ZeroMQ broker before workers so its sockets are ready when workers connect.
         # Skip if a broker is already running (e.g. started by the test fixture).
-        if self.profile.process_control_backend == 'core.zeromq':
-            from aiida.manage.manager import get_manager
+        from aiida.brokers.zeromq import ZeromqBroker
+        from aiida.manage.manager import get_manager
 
+        try:
             broker_instance = get_manager().get_broker()
-            broker_already_running = (
-                broker_instance is not None and hasattr(broker_instance, 'is_running') and broker_instance.is_running
+        except Exception:
```

Ruff (0.15.20)

[warning] Do not catch blind exception: `Exception`

(BLE001)